helm/chaos-mesh/templates/_helpers.tpl

```diff
@@ -156,3 +156,18 @@ Define the webhook's name
 {{- define "chaos-dlv.image" -}}
 {{ .Values.chaosDlv.image.registry | default .Values.images.registry }}/{{ .Values.chaosDlv.image.repository }}:{{ .Values.chaosDlv.image.tag | default .Values.images.tag }}
 {{- end -}}
+
+{{/*
+Return the appropriate apiVersion for ingress
+*/}}
+{{- define "chaos-dashboard.ingress.apiVersion" -}}
+{{- if .Values.dashboard.ingress.apiVersionOverrides -}}
+{{- print .Values.dashboard.ingress.apiVersionOverrides -}}
+{{- else if semverCompare "<1.14-0" .Capabilities.KubeVersion.Version -}}
+{{- print "extensions/v1beta1" -}}
+{{- else if semverCompare "<1.19-0" .Capabilities.KubeVersion.Version -}}
```

Comment:
I had issues with this on eks, eks returns `1.21.9-eks-0d102a7` for the version and the semverCompare doesn't properly resolve.

Can you maybe use `Capabilities.KubeVersion.Major`?

Reply:
Or at least allow for a manual override with something like this: https://github.com/openfaas/faas-netes/blob/master/chart/openfaas/templates/_helpers.tpl#L22-L25

Reply:
@MarkAndersonTrocme 
Thank you for the review.
As you reviewed, I added `kubeVersionOverride` to the `values.yaml` and configured it to override the kubernetes version.

Reply:
Great, @STRRL just waiting on your review then 😄

Reply:
Hi @KimDoubleB, I prefer to use `Capabilities.APIVersions.Has` with `networking.k8s.io/v1`, then fallback to `networking.k8s.io/v1beta1`, at the last fallback to `extensions/v1beta1"`

Reply:
Hi @KimDoubleB, I prefer to use `Capabilities.APIVersions.Has` with `networking.k8s.io/v1`, then fallback to `networking.k8s.io/v1beta1`, at the last fallback to `extensions/v1beta1"`

Reply:
rest LGTM ! PTAL @KimDoubleB 

cc @MarkAndersonTrocme

Reply:
@STRRL 
Thank you for the review. I modified the code. 
It's nice to see that it's more concise using `Capabilities.APIVersions.Has`.

Reply:
But, if you use `helm template` command, you must use option `--validate` (I used helm v3.9.0).